### PR TITLE
Margo/marco/sonarkt 337

### PR DESCRIPTION
**Pull request by kaufco**
Add a GradleSensor that will visit .gradle.kts files

### CODE REVIEW

File: sonar-kotlin-metrics/build.gradle.kts
Comment (johann-beleites-sonarsource) [inline sonar-kotlin-metrics/build.gradle.kts:1]: Hmm I see there are some TODO comments and such in this file, just as in `sonar-kotlin-gradle/build.gradle.kfs`. Is this intentional?

File: sonar-kotlin-metrics/build.gradle.kts [hunk 1]
```diff
@@ -2,6 +2,10 @@ plugins {
     kotlin("jvm")
 }
 
+repositories {
+    maven { url = uri("https://repo.gradle.org/gradle/libs-releases") }
```
Comment (johann-beleites-sonarsource) [inline sonar-kotlin-metrics/build.gradle.kts:6]: Do we need to add this repo in all submodules?

File: sonar-kotlin-plugin/build.gradle.kts [hunk 2]
```diff
@@ -8,6 +8,10 @@ plugins {
     kotlin("jvm")
 }
 
+repositories {
+    maven { url = uri("https://repo.gradle.org/gradle/libs-releases") }
```
Comment (johann-beleites-sonarsource) [inline sonar-kotlin-plugin/build.gradle.kts:12]: Do we need to add this repo in all submodules?

File: utils-kotlin/build.gradle.kts [hunk 3]
```diff
@@ -4,6 +4,10 @@ plugins {
     kotlin("jvm")
 }
 
+repositories {
+    maven { url = uri("https://repo.gradle.org/gradle/libs-releases") }
```
Comment (johann-beleites-sonarsource) [inline utils-kotlin/build.gradle.kts:8]: Do we need to add this repo in all submodules?

**Review by johann-beleites-sonarsource — COMMENTED** [review]
Did we consciously make the decision that Kotlin and Kotlin Gradle DSL are both treated as the same language?

File: README.md [hunk 4]
```diff
@@ -95,6 +110,49 @@ You can obviously do the same in the IDE and/or only run a particular test:
...
+
+**sonar-kotlin-checks:**
+
+- Rule implementations and tests go here.
+- Also contains helper functions if they are too specific for the `sonar-kotlin-api` layer,
+  but are rather implementation details of a rule.
+
+**sonar-kotlin-metrics:**
+
+- Implementations for certain code metrics such as cyclomatic complexity.
```
[the remark this answers is not in the document]
Comment (kaufco) [reply README.md:145]: Fixed.

File: README.md [hunk 5]
```diff
@@ -1,33 +1,48 @@
...
 * [Issue tracking](https://jira.sonarsource.com/browse/SONARKT/)
 * [Available rules](https://rules.sonarsource.com/kotlin)
 * [Community Forum](https://community.sonarsource.com/)
 
+## Building and Testing
+
 ### Build
 
-Build and run Unit Tests:
+Building and run Unit Tests:
```
[the remark this answers is not in the document]
Comment (kaufco) [reply README.md:41]: Fixed.

File: settings.gradle.kts [hunk 6]
```diff
@@ -45,13 +45,15 @@ dependencyResolutionManagement {
...
+            library("kotlin-scripting-compiler-embeddable", "org.jetbrains.kotlin", "kotlin-scripting-compiler-embeddable").version(kotlinVersion)
             library("sonar-analyzer-commons", "org.sonarsource.analyzer-commons", "sonar-analyzer-commons").versionRef(analyzerCommons)
             library("sonar-analyzer-commons-recognizers", "org.sonarsource.analyzer-commons", "sonar-analyzer-recognizers").versionRef(analyzerCommons)
             library("sonar-performance-measure", "org.sonarsource.analyzer-commons", "sonar-performance-measure").versionRef(analyzerCommons)
             library("sonar-plugin-api", "org.sonarsource.api.plugin", "sonar-plugin-api").version(sonarPluginApi)
             library("sonar-regex-parsing", "org.sonarsource.analyzer-commons", "sonar-regex-parsing").versionRef(analyzerCommons)
             library("sonar-xml-parsing", "org.sonarsource.analyzer-commons", "sonar-xml-parsing").versionRef(analyzerCommons)
             library("staxmate", "com.fasterxml.staxmate", "staxmate").versionRef(staxmate)
+            library("gradle-tooling-api","org.gradle", "gradle-tooling-api").version("7.5.1")
```
[the remark this answers is not in the document]
Comment (kaufco) [reply settings.gradle.kts:56]: Fixed.

File: settings.gradle.kts [hunk 7]
```diff
@@ -98,3 +100,4 @@ dependencyResolutionManagement {
         }
     }
 }
+include("sonar-kotlin-gradle")
```
[the remark this answers is not in the document]
Comment (kaufco) [reply settings.gradle.kts:103]: Fixed.

File: sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/checks/ProviderGetOutsideTaskCheck.kt
[the remark this answers is not in the document]
Comment (kaufco) [reply sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/checks/ProviderGetOutsideTaskCheck.kt:1]: This is a dummy check to check the bindings for the rule implementation unit tests while we do not yet have an actual Kotlin Gradle DSL rule implemented.

[the remark this answers is not in the document]
Comment (kaufco) [reply sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/checks/ProviderGetOutsideTaskCheck.kt:1]: Note: replaced in SONARKT-340 PR with some other content.

**Comment by johann-beleites-sonarsource** [comment]
Can we close this PR @kaufco @leveretka ?

**Pull request closed without merging**